### PR TITLE
Add AuthRepository to abstract handling a token.

### PR DESCRIPTION
**Pull request by saneyuki**
- This `Repository` is inspired by repository pattern.
- This regards cookie as the kind of database.

### CODE REVIEW

**Comment by saneyuki** [comment]
@Constellation r?

**Comment by saneyuki** [comment]
I worry about:

- should this place `script/adopter/`?
  - should create a new directory?
- about naming

File: client/script/adopter/AuthRepository.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    getToken() {
+        return this._cookie.get(KEY_TOKEN);
+    }
+
+    /**
+     *  @param  {string}  token
+     *  @return {void}
+     */
+    setToken(token) {
+        this._cookie.set(KEY_TOKEN, token);
```
Comment (saneyuki) [inline client/script/adopter/AuthRepository.js]: nit: set 3rd `expire` option.

**Comment by Constellation** [comment]
@saneyuki 

> should this place script/adopter/?

Your concern seems reasonable. In `karen`, `adopter` means like `abstraction`. So in the future, almost all code will locate in `adopter` :P 
Currently I have no idea about good directory name... Do you have any idea?

> about naming

It looks nice to me. Hiding the cookie jar by repository pattern & calling it as `AuthRepository` is reasonable I think.

**Comment by saneyuki** [comment]
> Currently I have no idea about good directory name... Do you have any idea?

When there are many files in `adopter/`, we should create `adopter/driver/` or `adopter/repository/` :)

**Pull request merged**